DIPs/1NNN-WGB.md

```diff
@@ -0,0 +1,155 @@
+# Argument Ownership and Function Calls
+
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | XXXX                                                            |
+| Review Count:   | 0                                                               |
+| Author:         | Walter Bright                                                   |
+| Implementation: |                                                                 |
+| Status:         |                                                                 |
+
+
+## Abstract
+
+DIP25 and DIP1000 greatly improved the memory safety of passing
+references and pointers to functions by checking whether those
+pointers escaped a function or not. This implies that a container
+can safely pass a reference to its internals to a function, if that
+function doesn't escape it.
+
+But if the function is also passed another reference to the container,
+that reference can render invalid what the other reference refers to.
```

Comment:
It is unclear which reference is meant. Maybe labeling the references as reference (A) and reference (B) would help.